src/audio/OpenALAudioEngine.cpp

```diff
@@ -0,0 +1,488 @@
+#ifdef RE_USE_SOUND
+
+#include "OpenALAudioEngine.h"
+
+#include <algorithm>
+#include <array>
+#include <atomic>
+#include <mutex>
+#include <thread>
+
+#include <AL/al.h>
+#include <AL/alc.h>
+
+using namespace Audio;
+
+static std::vector<std::string> enumeratedDevices;
+
+// The mutex is there so that static members
+// can be accessed in a thread-safe manner
+static std::mutex mtx_enumeratedDevices;
+
+constexpr std::size_t OPENAL_SOURCES_POOL = 256;
```

Comment:
Don't know regoth naming convection, but it looks like macro name.

Reply:
Yeah the rationale is that that variable is essentially to be considered a macro, but let the compiler treat it as a variable. If that makes sense anyway.